meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start.bb

```diff
@@ -22,6 +24,8 @@ do_install() {
 	install -d ${D}${bindir}
 	install -m 0755 ${WORKDIR}/compose-apps-early-start ${D}${bindir}/
 	install -m 0755 ${WORKDIR}/compose-apps-early-start-recovery ${D}${bindir}/
+	install -m 0755 ${WORKDIR}/start-apps ${D}${bindir}/
+	install -m 0755 ${WORKDIR}/start-restorable-apps ${D}${bindir}/
```

Comment:
maybe we could/should just have one script deal with both rather than splitting this up into so many pieces. I don't have a strong opinion. Sort of a distro maintenance question.

Reply:
OK. I actually implemented four flavors of the given implementation, the way you suggested is one of them. The overall approach works regardless of the flavor, I decided to go with the separate service scripts because it makes reviewing easier and arguably easier to do further support.

The four implementation options I tried are:

1. Dedicated systemd service and corresponding dedicated script for the restorable Apps. Which of the services to trigger is determined by  service conditions (e.g. `ConditionPathExists=/var/sota/reset-apps`). The current implementation is intact at all.

2. The current PR version. A single systemd service, two use-case specific scripts as well as an entry script with `if else` statement

3. A single systemd service, a single script that includes implementation for both use-cases as well as `if else`.

4. A single systemd service and a single script for both use-cases. `if else` is in the `aklite-app` utility that knows how to run both type of apps.


I played with all four approached, and none of them is ideal, each of them has its pros and cons. Based on my experience with it and taking into account a need in further support and maybe some customization, I incline towards 2 or 3. It implies that `aklite-apps` will run only restorable apps while compose apps will be started by means of `docker compose` and a script, effectively, as it is now.